data/src/main/scala/ch.epfl.scala.index.data/elastic/package.scala

```diff
@@ -2,48 +2,91 @@ package ch.epfl.scala.index
 package data
 
 import model._
-
 import org.elasticsearch.common.settings.Settings
 import com.sksamuel.elastic4s._
 import source.Indexable
-
 import org.json4s._
+import JsonDSL._
+import org.json4s.native.JsonMethods._
 import org.json4s.native.Serialization
 import org.json4s.native.Serialization.{read, write}
 
 trait ProjectProtocol {
+
   implicit val formats = Serialization.formats(ShortTypeHints(List(
     classOf[Milestone],
     classOf[ReleaseCandidate],
     classOf[OtherPreRelease]
-  )))
+  ))) ++ List(ScopeSerializer, DependencySerializer)
+
   implicit val serialization = native.Serialization
 
   implicit object ProjectAs extends HitAs[Project] {
-    override def as(hit: RichSearchHit): Project = {
-      read[Project](hit.sourceAsString)
-    }
+
+    override def as(hit: RichSearchHit): Project = read[Project](hit.sourceAsString)
   }
 
   implicit object ProjectIndexable extends Indexable[Project] {
-    override def json(project: Project): String = {
-      write(project)
-    }
+
+    override def json(project: Project): String = write(project)
   }
+
```

Comment:
As far as I understand, they work only for case classes. Dependency isn't :(
When you have an idea how to simplify that - I would appreciate.